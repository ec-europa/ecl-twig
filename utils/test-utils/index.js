const Twig = require('twig');
const { format } = require('prettier');

const renderTwigFile = (file, options, cb) =>
  Twig.renderFile(file, options, (err, html) =>
    cb(err, format(html, { parser: 'html' }))
  );

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) =>
    Twig.renderFile(file, options, (err, html) => {
      if (err) return reject(err);

      const div = document.createElement('div');
      div.innerHTML = html.trim();

      return resolve(div.firstChild);
    })
  );

module.exports = {
  renderTwigFile,
  renderTwigFileAsNode,
};
