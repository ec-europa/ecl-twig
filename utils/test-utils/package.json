{
  "name": "@ecl-twig/test-utils",
  "version": "2.29.0",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
