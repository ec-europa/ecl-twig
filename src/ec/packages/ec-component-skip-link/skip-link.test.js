import path from 'path';
import { merge, renderTwigFileAsNode } from '@ecl-twig/test-utils';

describe('EC - Skip Link', () => {
  const template = path.resolve(__dirname, './skip-link.html.twig');
  const render = params => renderTwigFileAsNode(template, params);
  const defaultDataStructure = {
    label: '',
    href: '',
  };

  describe('Default', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      const options = merge(defaultDataStructure, {
        label: 'Skip to main content',
        href: '#top',
      });

      return expect(render(options)).resolves.toMatchSnapshot();
    });
  });

  test('renders correctly', () => {
    expect.assertions(1);
    return expect(render()).resolves.toMatchSnapshot();
  });
});
