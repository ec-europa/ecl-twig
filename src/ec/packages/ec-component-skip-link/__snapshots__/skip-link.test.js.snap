// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC - Skip Link Default renders correctly 1`] = `
<a
  class="ecl-skip-link"
  href="#top"
>
  Skip to main content
</a>
`;

exports[`EC - Skip Link renders correctly 1`] = `
<a
  class="ecl-skip-link"
  href=""
/>
`;
